unity/Holo/Assets/Scripts/STLConverter.cs

```diff
@@ -0,0 +1,178 @@
+            Debug.Log("Doing mesh #" + count.ToString() + "!");
+            stlFileImporter.Load_STL_file(file_paths[1], first_mesh);
+            mesh.AddBlendShapeFrame(Path.GetFileName(file_paths[i]), 100f, stlFileImporter.AllVertices, null, null);
+            first_mesh = false;
+            count++;
+        }
+
+        skinnedMesh.sharedMesh = mesh;
+    }
+}
+
+
+//Loads a single STL file and turns it into a list of vertices (x,y,z) & if first_mesh: a list of indexes
+public class STLFileImporter
+{
+    public Vector3[] BaseVertices { get; private set; }
+
+    private List<Vector3> allVertices = new List<Vector3>();
+    public Vector3[] AllVertices { get => allVertices.ToArray(); }
+
+    private List<int> indices = new List<int>();
+    public int[] Indices { get => indices.ToArray(); }
+
+    private uint facetCount = 1;
+
+    public void Load_STL_file(string file_path, bool first_mesh)
+    {
+        allVertices = new List<Vector3>();
+        using (FileStream filestream = new FileStream(file_path, FileMode.Open, FileAccess.Read))
+        {
+            using (BinaryReader binary_reader = new BinaryReader(filestream, new ASCIIEncoding()))
+            {
+                // read header
+                byte[] header = binary_reader.ReadBytes(80);
+                facetCount = binary_reader.ReadUInt32();
+
+                if (first_mesh)
+                {
+                    BaseVertices = new Vector3[facetCount*3];
+                }
+
+                //TODO check topology
+
+                for (uint i = 0; i < facetCount; i++)
+                    Adapt_Facet(binary_reader, first_mesh);
+            }
+        }
+        BaseVertices = new Vector3[allVertices.Count];
+    }
+
+    private void Adapt_Facet(BinaryReader binary_reader, bool first_mesh)
+    {
+        binary_reader.GetVector3(); // A normal we don't use
+
+        for (int i = 0; i < 3; i++)
+        {
+            Vector3 vertix = binary_reader.GetVector3();
+            AddVertix(vertix, first_mesh);
+        }
+        binary_reader.ReadUInt16(); // non-sense attribute byte
+    }
+
+    private void AddVertix(Vector3 vertix, bool first_mesh)
+    {
+        int index = 0;
+        foreach (Vector3 listed_vertix in allVertices)
+        {
+            if (listed_vertix.Equals(vertix))
+            {
+                if(first_mesh)
+                    this.indices.Add(index);
+                return;
+            }
+            index++;   
+        }
+        this.indices.Add(index);
+        allVertices.Add(vertix);
+
+    }
+}
+
+//Static class containing methods for extracting and adapting vertices from a STL file. 
+public static class STLImportUtils
+{
+    public static Vector3 GetVector3(this BinaryReader binaryReader)
+    {
+        Vector3 vector3 = new Vector3();
+        for (int i = 0; i < 3; i++)
+        {
+            vector3[i] = binaryReader.ReadSingle();
+            if (i == 2)
+                vector3[i] = -vector3[i];
+        }
```

Comment:
I would suggest it's a bit clearer to write `vector3.z = -vector3.z` outside of the `for` loop, instead of 

```
            if (i == 2)
                vector3[i] = -vector3[i];
```

inside the loop.